src/main/java/org/gooru/insights/services/ItemServiceImpl.java

```diff
@@ -391,6 +411,116 @@ private String appendQuery(String requestId, String prefix, List<Map<String, Obj
+
+		String delimiter = getBaseConnectionService().getExportReportCache().get(APIConstants.DELIMITER);
+		int defaultLimit = Integer.valueOf(getBaseConnectionService().getExportReportCache().get(APIConstants.DEFAULT_LIMIT));
+		int offSet = 0;
+		int totalRows = 0;
+		boolean isNewFile = true;
+		try {
+			
+			RequestParamsDTO requestParamsDTO = getBaseAPIService().buildRequestParameters(data);
+			RequestParamsPaginationDTO paginationDTO = requestParamsDTO.getPagination();
+			ResponseParamDTO<Map<String, Object>> responseDTO = null;
+			
+			offSet = paginationDTO.getOffset();
+			paginationDTO.setLimit(defaultLimit);
+			requestParamsDTO.setPagination(paginationDTO);
+			
+			if (userMap == null) {
+				userMap = getUserObjectData(traceId,sessionToken);
+			}
+
+			Map<String, Boolean> checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+			requestParamsDTO = getUserService().validateUserRole(traceId,requestParamsDTO, userMap);
+			String[] indices = getBaseAPIService().getIndices(requestParamsDTO.getDataSource().toLowerCase());
+			
+			do {
+				responseDTO = getEsService().generateQuery(traceId,requestParamsDTO, indices, checkPoint);
+				getCSVFileWriterService().generateCSVReport(new HashSet<String>(Arrays.asList(requestParamsDTO.getFields().split(APIConstants.COMMA))), responseDTO.getContent(), absoluteFilePath, delimiter, isNewFile);
+				checkPoint.put(Hasdatas.HAS_MULTIGET.check(), false);
+				/*Incrementing offset values */
+				offSet += defaultLimit;
+				paginationDTO.setOffset(offSet);
+				requestParamsDTO.setPagination(paginationDTO);
+				
+				checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+				totalRows = Integer.valueOf(responseDTO.getPaginate().get(APIConstants.TOTAL_ROWS).toString());
+				isNewFile = false;
+			} while(offSet < totalRows);
+			
+		} catch (Exception e) {
+			logger.error("Error while writting file. {}", e);
+		}
+	}
+	
+	@Override
+	public ResponseParamDTO<Map<String, Object>> exportReport(HttpServletResponse response, String traceId, String data, String sessionToken, Map<String, Object> userMap) {
+
+		int maxLimit = Integer.valueOf(getBaseConnectionService().getExportReportCache().get(APIConstants.MAX_LIMIT));
+		int totalRows = 0;
+		String absoluteFilePath = null;
+		String fileName = UUID.randomUUID().toString();
+		ResponseParamDTO<Map<String, Object>> responseDTO = null;
+		
+		try {
+			
+			RequestParamsDTO requestParamsDTO = getBaseAPIService().buildRequestParameters(data);
+			
+			if (userMap == null) {
+				userMap = getUserObjectData(traceId,sessionToken);
+			}
+
+			Map<String, Boolean> checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+			requestParamsDTO = getUserService().validateUserRole(traceId,requestParamsDTO, userMap);
+			String[] indices = getBaseAPIService().getIndices(requestParamsDTO.getDataSource().toLowerCase());
```

Comment:
By any chance the datasource will be null ? if so the null pointer exception will happen.